src/main/java/net/atos/entng/calendar/helpers/EventHelper.java

```diff
@@ -281,118 +288,202 @@ public void handle(Either<String, JsonObject> event) {
+                            if(message.getBoolean("sendNotif") == null || Boolean.FALSE.equals(isCreated)) {
+                                notifyUsersSharing(request, user, id.toString(), calendarEvent, isCreated, restrictedEvent);
+                            }
                         }
                     }
+
                 }
 
             }
         });
     }
 
+
     /**
      *
      * @param request
      * @param user
      * @param calendarId
      * @param calendarEvent
+     * @param restrictedEvent
      */
-    public void notifyUsersSharing(final HttpServerRequest request, final UserInfos user, final String calendarId, final JsonObject calendarEvent, final boolean isCreated ){
-        QueryBuilder query = QueryBuilder.start("_id").is(calendarId);
-        JsonObject keys = new JsonObject().put("calendar", 1);
-        JsonArray fetch = new JsonArray().add("shared");
+    public void notifyUsersSharing(final HttpServerRequest request, final UserInfos user, final String calendarId,
+                                   final JsonObject calendarEvent, final boolean isCreated, Boolean restrictedEvent) {
+        String collection;
+        QueryBuilder query;
+        JsonObject keys;
+        JsonArray fetch;
 
-        findRecipiants("calendar", query, keys, fetch, user, new Handler<Map<String, Object>>() {
-            @Override
-            public void handle(Map<String, Object> event) {
-                if (event != null) {
-                    List<String> recipients = (List<String>) event.get("recipients");
-                    String calendarTitle = (String) event.get("calendarTitle");
-                    if (recipients != null) {
-                        String template = isCreated ? "calendar.create" : "calendar.update";
-
-                        JsonObject p = new JsonObject()
-                                .put("uri",
-                                        "/userbook/annuaire#" + user.getUserId() + "#" + user.getType())
-                                .put("username", user.getUsername())
-                                .put("CalendarTitle", calendarTitle)
-                                .put("postTitle", calendarEvent.getString("title"))
-                                .put("profilUri",
+        collection = CALENDAR_COLLECTION;
+        query = QueryBuilder.start("_id").is(calendarId);
+        keys = new JsonObject().put("calendar", 1);
+        fetch = new JsonArray().add("shared");
+
+
+        findRecipiants(collection, query, keys, fetch, user, restrictedEvent, calendarEvent, event ->
+                sendNotificationToRecipients(event, isCreated, user, calendarEvent, calendarId, request));
+    }
+
+    private void sendNotificationToRecipients(Map<String, Object> event, boolean isCreated, UserInfos user, JsonObject calendarEvent, String calendarId, HttpServerRequest request) {
+        if (event != null) {
+            List<String> recipients = (List<String>) event.get("recipients");
```

Comment:
Tu pourrais faire event.get("recipients") dans au dessus de sorte à ce que tu appelles ou non sendNoficiation avec uniquement List<String> recipients

Reply:
Tu devrais avoir Unchecked cast: 'java.lang.Object' to 'java.util.List<java.lang.String>'